docs/spec.md

```diff
@@ -603,24 +603,25 @@ Performs element-wise addition of two tensors `lhs` and `rhs` and produces a
 
 | Label | Name  | Type                       | Constraints |
 |-------|-------|----------------------------|-------------|
-| (I1)  | `lhs` | tensor or quantized tensor | (C1-C4)     |
-| (I2)  | `rhs` | tensor or quantized tensor | (C1-C3)     |
+| (I1)  | `lhs` | tensor or quantized tensor | (C1-C5)     |
+| (I2)  | `rhs` | tensor or quantized tensor | (C1-C4)     |
 
 #### Outputs
 
 | Name     | Type                       | Constraints |
 |----------|----------------------------|-------------|
-| `result` | tensor or quantized tensor | (C1-C3)     |
+| `result` | tensor or quantized tensor | (C1-C4)     |
 
 #### Constraints
 
 * (C1) `shape(lhs) = shape(rhs) = shape(result)`.
 * If the operation uses non-quantized tensors:
   * (C2) `element_type(lhs) = element_type(rhs) = element_type(result)`.
 * If the operation uses quantized tensors:
-  * (C3) `element_type(lhs) = element_type(rhs) = element_type(result)`,
+  * (C3) `is_quantized_tensor(lhs) and is_quantized_tensor(rhs) and is_quantized_tensor(result)`.
+  * (C4) `quantized_element_type(lhs) = quantized_element_type(rhs) = quantized_element_type(result)`,
```

Comment:
Let's keep `element_type` here and leave potential tightening to #1556. Overall, I think it'd be okay to say that element_type can produce both regular and quantized element types - that'll be considerably more concise than the alternative.